### DISCUSSION TITLE
Does table endpoints support query params?

### DISCUSSION THREAD

**Discussion by am2222** [maplibre/martin#1413] [Q&A]
Hi,

I noticed function endpoints support a query param. I was wondering if we can add such to the table endpoint too? 

I think managing query params will be easier in table level in server than inside a sql function

**Comment by nyurik** [comment]
I think there was a few discussions on this, but I don't think its a good idea because we would essentially wrap all SQL functionality as some query language, e.g. `?foo=bar&baz=bug` -> `WHERE foo = 'bar' AND baz = 'bug'`, but then we would have to add all other things like `AND, OR, NOT, IN` support - and at the end of the day, it will be really bad for performance due to lack of table indexes. So instead, I would much rather let site developers decide which specific filtering / aggregation functionality they want to support, and create their own custom SQL functions that only support what they want users to do.

**Reply by am2222** [reply]
Thanks @nyurik !
I understand! It can get even more complicated as users might want to query data based on ranges and so on.

It would be nice to have a series of such example of queries so users can ge some idea how they develope their own queries

**Reply by nyurik** [reply]
Documentation PRs are always welcome 😀
